src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -641,12 +643,21 @@ void TemporalMemory::compute(const SDR &activeColumns, bool learn,
                              const SDR &extraWinners)
 {
   activateDendrites(learn, extraActive, extraWinners);
+
+  // Update Anomaly Metric.  The anomaly is the percent of active columns that
+  // were not predicted.
+  anomaly_ = nupic::algorithms::anomaly::computeRawAnomalyScore(
+                activeColumns,
+                cellsToColumns( getPredictiveCells() ));
```

Comment:
I do like this method for its simplicity! :+1:  allowing to compute inbetween the colute()'s calls is a nice trick. 

Please consider the following improvements: 
1. move activateDendrites, activateCells to protected: 
  - leaving compute() the only public API, as in other classes
  - otherwise, user can break the trick by
```
compute()
TM.anomaly //OK
activateCells() //moving to T+1
TM.anomaly //incorrect value, still from T
```

2. construct compute as 
```
compute():
  activateDendrites //#1
  anomaly
  activateCells 
  activateDendrites //#2
```
  - this would remove the `throws` from getPredictiveCells() (and we can remove all the code with manual call to activateDendrites() just before getPredCells)
  - it is still as efficient, as (assuming the compute is the only interface to change state) the consecutive calls to activateDendrites would return the short-path.

Reply:
> `activateDendrites //#2`

This call might need external predictive inputs (aka `extra`), which the user needs supply.

Reply:
> otherwise, user can break the trick by

Or we could provide a new public method `updateAnomaly` which the user can call if they choose to eschew the compute method in favor of calling the individual methods.

If I get to add statistical analysis of the anomaly (as built-in) then we will certainly need a separate method for the anomaly.